src/main/java/mate/academy/EventManager.java

```diff
@@ -1,19 +1,26 @@
 package mate.academy;
 
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+
 public class EventManager {
-    public void registerListener(EventListener listener) {
+    private final List<EventListener> listeners = new CopyOnWriteArrayList<>();
 
+    public void registerListener(EventListener listener) {
+        listeners.add(listener);
     }
 
     public void deregisterListener(EventListener listener) {
-
+        listeners.remove(listener);
     }
 
     public void notifyEvent(Event event) {
-
+        for (EventListener listener : listeners) {
+            listener.onEvent(event);
```

Comment:
Iterating over CopyOnWriteArrayList using an enhanced for loop can be less performant due to the creation of a new iterator for each element. If performance is a concern and the list is expected to be large, consider alternatives for managing the listeners list or using a regular Iterator explicitly.